src/scripts/mutual_activity.js

```diff
@@ -0,0 +1,189 @@
+const GROUP_SIMILAR_LABEL_SELECTOR = 'label#rollups';
+
+const MUTUAL_ACTIVITY_CLASS = 'xkit-mutual-activity';
+
+const MUTUAL_ACTIVITY_STORAGE_KEY = 'mutual_activity';
+const IS_ACTIVATED_STORAGE_KEY = `${MUTUAL_ACTIVITY_STORAGE_KEY}.isActivated`;
+const SHOW_GROUPED_NOTIFICATIONS_PREFERENCE_KEY = 'showGroupedNotifications';
+
+const nonMutualStyleElement = buildStyle(
+  `${NOTIFICATION_SECTION_SELECTOR} ${NOTIFICATION_SELECTOR}:not(${FOLLOWED_SELECTOR}, ${ROLLUP_SELECTOR}) {
+     display: none !important;
+  }
+
+  ${GROUP_SIMILAR_SECTION_SELECTOR} ${ROLLUP_CHECKBOX_SELECTOR} {
+    pointer-events: none;
+    cursor: not-allowed;
+  }
+
+  ${GROUP_SIMILAR_SECTION_SELECTOR} ${CHECKBOX_STATE_SELECTOR} {
+    background-color: rgba(var(--black), 0.40) !important;
+    cursor: not-allowed;
+  }
+
+  ${GROUP_SIMILAR_LABEL_SELECTOR} {
+    color: rgba(var(--black), 0.40) !important;
+  }
+
+  ${GROUP_SIMILAR_SECTION_SELECTOR} {
+    pointer-events: none;
+  }
+
+  ${GROUP_SIMILAR_SECTION_SELECTOR}::after {
+    content: "⚠️ Controlled by XKit Mutual Activity";
+  }
+`
+);
+
+const toggleRollupCheckbox = (shouldToggleCheckbox) => {
+  if (shouldToggleCheckbox) {
+    document.querySelector(ROLLUP_CHECKBOX_SELECTOR).click();
+    document.querySelector(APPLY_FILTERS_SELECTOR).click();
+  } else {
+    document.querySelector(CLOSE_FILTER_DIALOG_SELECTOR).click();
+  }
+};
+
+const getRollupCheckboxState = () =>
+  document.querySelector(CHECKBOX_STATE_SELECTOR).childElementCount > 0;
+
+const isFilterDialogOpen = () =>
+  document.querySelector(GLASS_CONTAINER_DIALOG_SELECTOR) !== null;
+
+const toggleGroupedNotifications = async () => {
+  const { showGroupedNotifications } = await getPreferences(
+    MUTUAL_ACTIVITY_STORAGE_KEY
+  );
+  const filterButton = document.querySelector(FILTER_BUTTON_SELECTOR);
+
+  if (!isFilterDialogOpen()) {
+    filterButton.click();
+  }
+
+  toggleRollupCheckbox(getRollupCheckboxState() !== showGroupedNotifications);
+};
+
+const setIsActivated = async (isActivated) => {
+  await browser.storage.local.set({ [IS_ACTIVATED_STORAGE_KEY]: isActivated });
+};
+
+const getIsActivated = () =>
+  browser.storage.local.get(IS_ACTIVATED_STORAGE_KEY);
+
+const enableFilter = () => {
+  document.head.append(nonMutualStyleElement);
+
+  const notificationSection = document.querySelector(
+    NOTIFICATION_SECTION_SELECTOR
+  );
+  const scrollEvent = new Event('scroll', {
+    bubbles: true
+  });
+
+  notificationSection.dispatchEvent(scrollEvent);
+  toggleGroupedNotifications();
+  setIsActivated(true);
+};
+
+const disableFilter = () => {
+  nonMutualStyleElement.remove();
+  toggleGroupedNotifications();
+  setIsActivated(false);
+};
+
+const toggleFilter = async ({ target }) => {
+  target?.checked ? enableFilter() : disableFilter();
+};
+
+const isOnActivityPage = () =>
+  document.querySelector(FILTER_BUTTON_SELECTOR) !== null;
+
+const createToggleButton = async () => {
```

Comment:
```suggestion
const createToggleButton = async ([filterButton]) => {
```

Now you can take an array of page modifications in the callback!

Reply:
NICE! Now we can keep all the jQuery out of this script.